ocean_lib/assets/trusted_algorithms.py

```diff
@@ -26,24 +28,41 @@ def generate_trusted_algo_dict(
     }
     ```
     """
-    if isinstance(asset_or_did, V3Asset):
+    if isinstance(asset_or_did, (V3Asset, V4Asset)):
         ddo = asset_or_did
     else:
         ddo = resolve_asset(asset_or_did, metadata_cache_uri=metadata_cache_uri)
 
     algo_metadata = ddo.metadata
-    return {
-        "did": ddo.did,
-        "filesChecksum": create_checksum(
-            algo_metadata.get("encryptedFiles", "")
-            + json.dumps(algo_metadata["main"]["files"], separators=(",", ":"))
-        ),
-        "containerSectionChecksum": create_checksum(
-            json.dumps(
-                algo_metadata["main"]["algorithm"]["container"], separators=(",", ":")
-            )
-        ),
-    }
+    if ddo.version.startswith("4."):
```

Comment:
you can add these as typings in the asset, e.g. generate_trusted_algos. This "if" can be removed.